app/src/main/java/com/asfoundation/wallet/subscriptions/UserSubscriptionsLocalData.kt

```diff
@@ -0,0 +1,74 @@
+package com.asfoundation.wallet.subscriptions
+
+import com.appcoins.wallet.bdsbilling.SubscriptionSubStatus
+import com.appcoins.wallet.bdsbilling.UserSubscriptionResponse
+import com.appcoins.wallet.bdsbilling.subscriptions.*
+import com.asfoundation.wallet.subscriptions.db.UserSubscriptionEntity
+import com.asfoundation.wallet.subscriptions.db.UserSubscriptionsDao
+import io.reactivex.Observable
+import io.reactivex.Single
+
+class UserSubscriptionsLocalData(private val userSubscriptionsDao: UserSubscriptionsDao) {
+
+  fun insertSubscriptions(responseList: List<UserSubscriptionResponse>,
+                          walletAddress: String) {
+    return userSubscriptionsDao.insertSubscriptions(mapToEntity(responseList, walletAddress))
+  }
+
+  fun getSubscriptions(walletAddress: String, subscriptionSubStatus: SubscriptionSubStatus? = null,
+                       limit: Int? = null): Observable<UserSubscriptionsListResponse> {
+    return buildGetSubscriptionQuery(walletAddress, subscriptionSubStatus, limit)
+        .map { mapToResponse(it) }
+        .toObservable()
+  }
+
+  private fun buildGetSubscriptionQuery(walletAddress: String,
+                                        subscriptionSubStatus: SubscriptionSubStatus?,
+                                        limit: Int?): Single<List<UserSubscriptionEntity>> {
+    return if (subscriptionSubStatus != null && limit != null) {
```

Comment:
We can use `when` and avoid the comparison for the 2 variable in all the stages
first check `subscriptionSubStatus != null && limit != null`
then check `subscriptionSubStatus != null`
afterwards` limit != null`
and then the `else`